### PR TITLE
using bullet to reduce the number of querry and increase application …

### PR DESCRIPTION
**Pull request by pdkproitf**
…performance and fix comment if not have user

### CODE REVIEW

File: app/views/books/show.html.erb [hunk 1]
```diff
@@ -80,10 +80,14 @@
                           </div>
                         </div>
                         <div id="commented">
-                          <%= render partial: @book.comments %>
+                          <%= render partial: @book.comments.includes(:user) %>
```
Comment (at-dypd) [inline app/views/books/show.html.erb]: đưa includes vào controller

File: Gemfile [hunk 2]
```diff
@@ -57,5 +58,5 @@ group :development do
   gem 'web-console', '~> 2.0'
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
-
+  gem 'bullet' #The Bullet gem is designed to help you increase your application's performance by reducing the number of queries it makes. It will watch your queries while you develop your application and notify you when you should add eager loading (N+1 queries), when you're using eager loading that isn't necessary and when you should use counter cache
```
Comment (yoyocoder) [inline Gemfile]: you don't need to add the comment in gem file. it is unnecessary.

File: app/views/books/show.html.erb [hunk 3]
```diff
@@ -8,15 +8,15 @@
...
-            <img src="<%= book.photo %>" class="recent-thumb" alt="">
-            <h2><a href=<%= book_path(book) %>> <%= book.title %> </a></h2>
-            <div class="product-sidebar-price">
-              <ins>$<%= (book.cost - book.cost * book.sale).round(2) %></ins> <del>$<%= book.cost %></del>
+            <% @books.each do |book| %>
+            <div class="thubmnail-recent">
+              <img src="<%= book.photo %>" class="recent-thumb" alt="">
+              <h2><a href=<%= book_path(book) %>> <%= book.title %> </a></h2>
+              <div class="product-sidebar-price">
+                <ins>$<%= (book.cost - book.cost * book.sale).round(2) %></ins> <del>$<%= book.cost %></del>
```
Comment (yoyocoder) [inline app/views/books/show.html.erb]: `(book.cost - book.cost * book.sale)` you should move it to helper or model

File: Gemfile.lock [hunk 4]
```diff
@@ -1,237 +0,0 @@
...
-  sdoc (~> 0.4.0)
-  shoulda-matchers
-  spring
-  turbolinks
-  uglifier (>= 1.3.0)
-  web-console (~> 2.0)
-  will_paginate
-
-BUNDLED WITH
-   1.12.5
```
Comment (yoyocoder) [inline Gemfile.lock]: why did you delete Gemfile.lock? 😱

Comment (pdkproitf) [inline Gemfile.lock]: because it make me can't  push the branch to githup.

Comment (pdkproitf) [inline Gemfile.lock]: Gemfile.lock wil autogenous when i run server. i think it not necessary

File: Gemfile (hunk 2 above)
Comment (pdkproitf) [inline Gemfile]: yep . i comment because i want to understand what is that gem will to do? if i have review code in feature. feature it's mean long time after.
